### DISCUSSION TITLE
Internal support for mongodb

### DISCUSSION THREAD

**Discussion by A-cute-blob** [Androz2091/discord-giveaways#382] [General]
Maybe add mongodb as a storage option. It will be really good

**Comment by Nico105** [comment]
https://github.com/Androz2091/discord-giveaways/tree/master/examples/custom-databases

**Comment by DragonZV** [comment]
you should add it as a builtin option

**Comment by Nico105** [comment]
... I already told you arguments against it, but talk to Androz if you're that convinced of it

EDIT: 
-  goes over basic functionality of the module since JSON is default anyway.
- we have many other db examples too=would we need to support them internally too or what?
- we would need to add the db packages as dependencies=unnecessary size
   - and when we would use "try catch" to not have them as dependencies then it would not feel right/leave room for errors.
- it would just go out of hand, see point 1

PRO argument: we could make sure that the dbs work as they should=user cant make too many mistakes
CONTRA for the same: much responsibility/we would be at fault for many errors, even though it isnt needed for the basic functionality.

But thats just my opinion.
Feel free to try to argue, but I don't see it as such a good idea.

**Reply by Shashank3736** [reply]
Creating a MongoDB option is no good but JSON is not actually a database so using sqlite3 as a default database will be a better option.

**Comment by A-cute-blob** [comment]
> you should add it as a builtin option

thats what i mean kek

**Comment by twlite** [comment]
in my opinion, adding mongodb or any other database internally to this package would be a bad idea.

Today, you are saying "mongodb", tomorrow someone else will ask for "postgres", then x and y and it continues. It will make the package heavy and will require maintainers to keep it up to date with database clients just for "discord giveaways" storage.

There are several examples already on this https://github.com/Androz2091/discord-giveaways/tree/master/examples/custom-databases which shows how to use discord-giveaways

Maybe you have points like `but I only said "mongodb", not others`. Yes you are right, but what about the people who dont want to use mongodb? They will be forced to have additional dependencies for absolutely no reason.

Current implementation is the best one. You have the freedom of choice, use whatever you prefer with this package.
